Scripts/RTS/Chunk.cs

```diff
@@ -1,32 +1,47 @@
 ﻿namespace RTS;
+using System.Collections.Generic;
 
 public class Chunk
 {
     TileMap tileMap;
 
-    public Chunk(int chunkX, int chunkY, TileMap tileMap, Noise noise)
+    public Chunk(int chunkX, int chunkY, TileMap tileMap, Noise noise, bool generate = true)
     {
         this.tileMap = tileMap;
+        if(generate) this.GenerateWorld(chunkX, chunkY, tileMap, noise);
+    }
 
+    void GenerateWorld(int chunkX, int chunkY, TileMap tileMap, Noise noise)
+    {
         for (int x = 0; x < World.ChunkSize; x++)
         {
             for (int y = 0; y < World.ChunkSize; y++)
             {
                 var globalX = (chunkX * World.ChunkSize) + x;
                 var globalY = (chunkY * World.ChunkSize) + y;
 
-                string type;
+                string type = "";
+                var currentNoise = noise.GetNoise2D(globalX, globalY);
+                var name = ((string) (this.tileMap.Name)).ToLower();
```

Comment:
Okay now that I'm actually debugging your code I see what your intent here was. You assume every tile will have "grass" in the name. This could work I guess? Like there could be a tile called "grass_dirt_1" or "grass_cobblestone_23". But this is a bit odd to me to do it this way. There has to be another way to do this.